### ISSUE TITLE
Payment method is not included in paymentMethodsResponse, but component is still displayed.

### ISSUE THREAD

**Issue by gndk** [Adyen/adyen-web#1048]
**Describe the bug**
Payment method is not included in `paymentMethodsResponse`, but component is still displayed.

**To Reproduce**
We followed the exact steps in the docs for integrating the giropay payment method: https://docs.adyen.com/payment-methods/giropay/web-component#show-giropay-in-your-payment-form

Everything works fine for `countryCode: DE` and `amount.currency: EUR`. The giropay method is included in `paymentMethodsResponse` and the payment process works as expected.

If we pass a different value to the `/paymentMethods` endpoint, like `countryCode: US` and `amount.currency: USD`, the giropay method is not included in the `paymentMethodsResponse`, as expected.

However, then component is still rendered / giropay button displayed.

**Expected behavior**
Component will not be displayed if the payment method is not included in `paymentMethodsResponse`.

**Additional context**
The docs for components integration at https://docs.adyen.com/online-payments/components-web#step-1-get-available-payment-methods say:

> Pass the response to your front end. You'll use this in the next step to show which payment methods are available for the shopper.

This sounds like the Adyen Web package will handle the conditional displaying of the component automatically for us, based on the `paymentMethodsResponse` that we pass when creating an instance of `AdyenCheckout`.

I am not sure if:

1. this assumption is wrong and we need to handle this logic ourselves
2. we are missing something in our current implementation
3. this is a possible bug

If option 1 does not apply, I'll be happy to provide additional details to narrow it down.

**Comment by marcperez** [comment]
Hi @gndk,

On the Drop-in integration, the payment method list will show only the payment methods available on the `paymentMethodsResponse`. On the Components integration, on the other hand, you're in control of which payment method components are displayed.
We do provide a helper that might be useful here, and  that you could use to conditionally `mount()` components:

```js
const checkout = new AdyenCheckout({ 
  paymentMethodsResponse: paymentMethodsResponse
  /* ... */ 
});

checkout.paymentMethodsResponse.has('ideal'); // => true/false
```

Hope this helps!

**Comment by gndk** [comment]
Hi @marcperez 

Thanks for the quick response!

I actually found that helper method while reading through the library files and tried to use it. But I was obviously doing it wrong, because I tried stuff like `configuration.paymentMethodsReponse.has('giropay')` and got all kinds of errors.

Don't know why I didn't occur to me to use it on the `checkout`. Works like a charm :-)

Might be a useful addition to the components documentation. Here at https://docs.adyen.com/online-payments/components-web#step-2-add-components in step 5 were you have the example:

![image](https://user-images.githubusercontent.com/25748293/117648073-132b2880-b18e-11eb-9a3c-b6f50767bce3.png)
